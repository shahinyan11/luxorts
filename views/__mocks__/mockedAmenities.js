export default {
  'Amina Heaney': {
    '5e066ad7-4194-4bd4-95ec-0d9715f8159b': {
      id: '5e066ad7-4194-4bd4-95ec-0d9715f8159b',
      description: 'Maiores porro nostrum dolorum sunt.',
      name: 'Nolan Kertzmann',
      status: 'active',
      category: null,
      checked: false,
    },
  },
  'Torrie Sawayn': {
    'c277033f-1683-4d2d-a409-d00b0bbe8201': {
      id: 'c277033f-1683-4d2d-a409-d00b0bbe8201',
      description: 'Nemo magnam dolores quam voluptatibus distinctio unde.',
      name: 'Stephine Rohan',
      status: 'active',
      category: null,
      checked: false,
    },
  },
};
