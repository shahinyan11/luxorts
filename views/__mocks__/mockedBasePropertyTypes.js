export default [
  {
    value: '51e35121-c27e-46fc-9feb-b51dab2490ee',
    label: 'repudiandae',
    description: 'description repudiandae',
  },
  {
    value: '995c5714-185d-4085-8512-690903334f6e',
    label: 'molestias',
    description: 'description molestias',
  },
];
