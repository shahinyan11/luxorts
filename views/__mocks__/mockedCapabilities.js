export default {
  APPROVED: {
    '11da7909-9f4c-45d6-a2a0-8a105b067ed2': {
      id: '11da7909-9f4c-45d6-a2a0-8a105b067ed2',
      description: 'Vel commodi ad incidunt repellendus a molestias delectus nam.',
      name: 'Tonya Wunsch',
      status: 'active',
      checked: false,
    },
    'b84c898e-2176-4ac8-9dfa-54186ccef6cb': {
      id: 'b84c898e-2176-4ac8-9dfa-54186ccef6cb',
      description: 'Maxime nostrum ex iste velit nulla incidunt.',
      name: 'Edwin Greenholt',
      status: 'active',
      checked: false,
    },
  },
};
