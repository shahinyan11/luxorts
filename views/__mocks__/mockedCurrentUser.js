export default {
  id: '5c5d814c-8e06-433d-b006-4119311585bd',
  email: 'evhenii.hryhoriev+10@rubygarage.org',
  publicId: 'c4toxHVi',
  verifiedStatus: 'unverified',
  userType: 'traveller_account',
  userProfile: {
    id: '52080e19-c5e0-45bb-af58-edbbe70e78fa',
    about: null,
    address: null,
    avatarUrls: {
      original: '/avatar-img-url',
    },
    dateOfBirth: '1994-01-01',
    drivingLicenseFirstName: 'Evhenii',
    drivingLicenseLastName: 'Test',
    firstName: 'Evhenii',
    gender: null,
    lastName: 'Test',
    phoneNumber: '+299999999',
    street: 'street',
    city: 'city',
    state: 'state',
    zipCode: 'zipCode',
    country: 'country',
  },
  userInvitation: [
    {
      id: '52080e19-c5e0-45bb-af58-edbbe70e78fa',
      created_at: '2022-06-23T06:32:55.588Z',
      email: '12@mail.ru',
      last_sent_at: '2022-06-23T06:32:55.474Z',
      resendable: true,
      status: 'pending',
      user_account_id: '7211ccc2-1fca-4799-b1e1-4005e9fc42be',
    },
  ],
};
