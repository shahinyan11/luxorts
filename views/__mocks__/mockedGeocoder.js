export default [
  {
    streetNumber: '240',
    street: '240 North 7th Avenue',
    city: 'Phoenix',
    state: 'Arizona',
    country: 'United States',
    zipCode: '85007',
    fullAddress: '240 N 7th Ave, Phoenix, AZ 85007, USA',
    longitude: -112.0825682,
    latitude: 33.4507026,
  },
];
