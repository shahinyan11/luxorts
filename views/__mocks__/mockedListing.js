export default {
  id: '0645baf3-945d-4122-8a0b-5ab3e95a9b3e',
  applicationStep: 'accomodation',
  status: 'in_draft',
  userAccount: { id: '6cbf2b68-4a26-41b5-bcd5-161cc37e1420', type: 'hostAccount' },
  listingPropertyType: {
    id: 'ea19f127-bc89-4799-bdbb-346126ac3845',
    typeOfPlace: 'private_place',
    propertyType: {
      id: '51e35121-c27e-46fc-9feb-b51dab2490ee',
      name: 'repudiandae',
      description: 'description repudiandae',
    },
  },
  listingLocation: {
    id: '2b4032ec-a469-4489-b7dc-536d8f9552eb',
    country: 'United States',
    city: 'Phoenix',
    state: 'Arizona',
    street: 'West Maryland Avenue',
    apartmentNumber: '12',
    zipCode: '12345',
    longitude: '-73.97447621784085',
    latitude: '40.764468117826794',
  },
  listingDescription: null,
  listingPricing: {
    pricePerDay: 64,
  },
};
