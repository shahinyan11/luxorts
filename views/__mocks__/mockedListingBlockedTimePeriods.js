export default ['25/07/2022', '26/07/2022', '27/07/2022'];
