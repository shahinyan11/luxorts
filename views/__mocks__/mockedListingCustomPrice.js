export default {
  '01/09/2022': 64,
  '02/09/2022': 64,
  '03/09/2022': 64,
  '04/09/2022': 64,
  '05/09/2022': 64,
};
