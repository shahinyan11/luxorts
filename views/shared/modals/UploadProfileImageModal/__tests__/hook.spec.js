import { act, renderHook } from '@testing-library/react-hooks';

import { AVATAR_UPLOAD_BOUNDARIES, PHOTOS_VALIDATION } from 'constants/validations';

import { userUpdateAvatar } from 'state/concepts/session/actions';
import { MESSAGE_TYPE } from 'state/flash-messages/messagesTypes';
import { showMessage } from 'state/flash-messages/actions';
import { dispatch } from '__mocks__/react-redux';

import useContainer from '../hook';

jest.mock('utils/users/getCroppedImg', () =>
  jest.fn(() => ({
    type: 'image/jpeg',
  })),
);

const mockFile = (size) => {
  global.File = class MockedFile {
    constructor() {
      this.size = size;
    }
  };
};

describe('UploadProfileImageModal useContainer hook', () => {
  let result = null;
  const props = { initialImgSrc: 'initialImgSrc' };

  beforeAll(() => {
    window.URL.createObjectURL = jest.fn((fileUrl) => fileUrl);
  });

  afterAll(() => {
    window.URL.createObjectURL.mockReset();
  });

  beforeEach(() => {
    ({ result } = renderHook(() => useContainer(props)));

    jest.clearAllMocks();
  });

  it('matches snapshot', () => {
    expect(result.current).toMatchSnapshot();
  });

  it('onImageLoad method should updates crop state and center crop region considering to image width and height', () => {
    const { crop } = result.current;
    const event = {
      currentTarget: {
        width: 1000,
        height: 1000,
      },
    };
    const { width, height } = event.currentTarget;
    const minSize = Math.min(width, height, AVATAR_UPLOAD_BOUNDARIES.DEFAULT_CROP_DIAMETER);
    const delta = 20;

    const expected = {
      ...crop,
      width: minSize - delta,
      height: minSize - delta,
      x: width / 2 - (minSize - delta) / 2,
      y: height / 2 - (minSize - delta) / 2,
      image: event.currentTarget,
    };

    act(() => {
      result.current.onImageLoad(event);
    });

    expect(result.current.crop).toEqual(expected);
  });

  it('onCropChange method should updates crop state with new crop data', () => {
    act(() => {
      result.current.onCropChange('foo');
    });

    expect(result.current.crop).toEqual('foo');
  });

  it('onFileChange method should updates imgSrc state', () => {
    act(() => {
      result.current.onFileChange({ file: { originFileObj: 'originFileObj' } });
    });

    expect(result.current.imgSrc).toBe('originFileObj');
  });

  describe('onSave method', () => {
    it('should shows error when imgSrc isn`t present', () => {
      ({ result } = renderHook(() => useContainer({})));

      result.current.onSave();

      expect(dispatch).toHaveBeenCalledWith(
        showMessage({
          messageType: MESSAGE_TYPE.ERROR,
          description: { id: 'validations.pleaseUploadTheImage' },
        }),
      );
    });

    it('should dispatches userUpdateAvatar action with file', async () => {
      mockFile(PHOTOS_VALIDATION.MAX.SIZE);

      await result.current.onSave();

      expect(dispatch).toHaveBeenCalledWith(userUpdateAvatar({ size: PHOTOS_VALIDATION.MAX.SIZE }));
    });

    it('should shows error when file.size large than allowed', async () => {
      mockFile(PHOTOS_VALIDATION.MAX.SIZE + 1);

      await result.current.onSave();

      expect(dispatch).toHaveBeenCalledWith(
        showMessage({
          messageType: MESSAGE_TYPE.ERROR,
          description: { id: 'validations.avatarValidationError' },
        }),
      );
    });
  });
});
