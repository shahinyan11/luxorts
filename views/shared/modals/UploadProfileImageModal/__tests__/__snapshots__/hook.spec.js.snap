// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadProfileImageModal useContainer hook matches snapshot 1`] = `
Object {
  "crop": Object {
    "height": 240,
    "image": null,
    "unit": "px",
    "width": 240,
    "x": 0,
    "y": 0,
  },
  "imgSrc": "initialImgSrc",
  "isLoading": false,
  "onCropChange": [Function],
  "onFileChange": [Function],
  "onImageLoad": [Function],
  "onSave": [Function],
}
`;
